assets/src/edit-story/app/media/media3p/typedefs.js

```diff
@@ -0,0 +1,186 @@
+ * @property {string} poster
+ * @property {string} posterId
+ * @property {Object} sizes
+ * @property {string} src
+ * @property {string} title
+ * @property {string} typedef
+ */
+
+/**
+ * media3p/[provider] typedefs =================================================
+ */
+
+/**
+ * 'media3p/[provider]' action typedef
+ *
+ * @typedef {(payload: {provider: string, id: string}) => undefined} DeleteMediaElement
+ * @typedef {(payload: {provider: string}) => undefined} FetchMediaError
+ * @typedef {(payload: {provider: string, pageToken: string}) => undefined} FetchMediaStart
+ * @typedef {(payload: {provider: string, media, nextPageToken: string, totalPages: number}) => undefined} FetchMediaSuccess
+ * (Can take extra properties in the payload)
+ * @typedef {(payload: {provider: string}) => undefined} SetNextPage
+ * @typedef {(payload: {provider: string, id: string }) => undefined} UpdateMediaElement
+ * (Can take extra properties in the payload)
+ */
+
+/**
+ * 'media3p/[provider]' actions exposed through context providers
+ *
+ * @typedef {Object} Media3pProviderContextActions
+ * @property {SetNextPage} setNextPage action to set next page
+ * @property {import('./categories/typedefs').SelectCategory} selectCategory action to select category
+ * @property {import('./categories/typedefs').DeselectCategory} deselectCategory action to deselect current
+ * category
+ */
+
+/**
+ * 'media3p/[provider]' state object used by each provider, exposed to the rest
+ * of application through context providers
+ *
+ * @typedef {Object} Media3pProviderContextState
+ * @property {CategoriesState} categories Category state object
+ * @property {boolean} hasMore has more media
+ * @property {boolean} isMediaLoaded is media loaded
+ * @property {boolean} isMediaLoading is media loading
+ * @property {Array.<Media>} media array of media
+ * @property {string} pageToken page token for current page
+ * @property {string} nextPageToken the page token for the next page
+ * @property {number} totalPages total pages
+ */
+
+/**
+ * 'media3p/[provider]' state object used by each provider internally by the reducer.
+ *
+ * @typedef {Object} Media3pProviderReducerState
+ * @property {CategoriesState} categories Category state object
+ * @property {boolean} hasMore has more media
+ * @property {boolean} isMediaLoaded is media loaded
+ * @property {boolean} isMediaLoading is media loading
+ * @property {Array} media array of media
+ * @property {string} pageToken page token for current page
+ * @property {string} nextPageToken the page token for the next page
+ * @property {number} totalPages total pages
+ */
+
+/**
+ *
+ * @typedef {Object} Media3pProviderContext
+ * @property {Media3pProviderContextState} state state of a single provider
+ * @property {Media3pProviderContextActions} actions actions for the provider
+ */
+
+/**
+ * media3p/ typedefs ===========================================================
+ */
+
+/**
+ * 'media3p' root action typedefs
```

Comment:
'media3p action typedefs not specific to any provider'?

Reply:
Done